test/models/utils/test_inducing_point_allocators.py

```diff
@@ -262,12 +288,29 @@ def test_inducing_points_different_when_minimizing(self):
 
 class TestPivotedCholeskyInit(BotorchTestCase):
     def test_raises_for_quality_function_with_invalid_shape(self):
-        with self.assertRaises(ValueError):
-            inputs = torch.rand(15, 1, device=self.device)
+        inputs = torch.rand(15, 1, device=self.device)
+        with torch.no_grad():
             train_train_kernel = (
                 MaternKernel().to(self.device)(inputs).evaluate_kernel()
             )
-            quality_scores = torch.ones([10, 1], device=self.device)
+        quality_scores = torch.ones([10, 1], device=self.device)
+        with self.assertRaises(ValueError):
```

Comment:
It'd be good explicitly check for the specific error using `assertRaisesRegex`.

Reply:
definitely agree. I ignored it since this test was pre-existing (I just rearranged it). Changing a bunch of these to `assertRaisesRegex` could be a good starter task